### PR TITLE
Final project

### PR DESCRIPTION
**Pull request by techmoves**
Hi,

This PR is raised to present the submission of the `Space Travelers' Hub` web app : 

https://space-travelers-hub-vls7.onrender.com 

This project is prepared by @mailsg and @techmoves

The project consists of a web application for a fictitious company that provides commercial and scientific space travel services. The application will allow users to book rockets and join selected space missions. 

The Space Travelers' Hub consists of **Rockets**, **Missions**, and the **My Profile** section.

**Rockets** : The Rockets section displays a list of all available SpaceX rockets. Users can book each rocket by clicking the reservation button or cancel the previously made booking.

**Missions** : The Missions section displays a list of current missions along with their brief description and participation status. There is also a button next to each mission that allows users to join the selected mission or leave the mission the user joined earlier.

**My Profile** : The My Profile section displays all reserved rockets and space missions.

`Config & basic setup` - 
- Created a React App.
- Installed `React Redux`, `Redux Logger` and `React Router`.
- Created **routes** and **view** components: `rockets`, `missions`, `my profile`.
    - Used `<NavLink />` for the page navigation links and style active class to indicate which section/page user is currently on (underline active navigation link). 
- Created directories for all Redux state slice files: `rockets`, `missions`

`Redux: Fetch data and update Redux store` -
- Upon first render fetched data from the SpaceX API endpoints: 
   - Rockets: `https://api.spacexdata.com/v4/rockets`
   - Missions: `https://api.spacexdata.com/v3/missions` 
- Once the data are fetched, dispatched an action to store the selected data in Redux store:
   - Rockets: `id`, `rocket_name`, `description`, `flickr_images`
   - Missions: `mission_id`, `mission_name`, `description`

**Render UI:lists**
- Used `useSelector()` Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes
- Rendered a list of rockets (as per design).
- Rendered a table with the missions' data (as per design).

**Redux: Written actions and reducers for booking rockets and joining missions**
- When a user clicks the "Reserve rocket" button action is dispatched to update the store.
- Created a reducer and action dispatcher for the "Join Mission" button. `reserved: true`

**Redux: Written actions and reducers for canceling rockets and leaving missions**
- set the `reserved` key to `false`.

**Render UI: conditional components rendering**
- Rockets which are reserved, show a `reserved` badge and `Cancel Reservation` to set to default `Reserve Rocket`
- Same with Missions. 

**Render UI: My Profile section**
- Composed two column layout and list ONLY the rockets reserved and missions joined by the user (as per design)
- Render a list of all joined missions and reserved rockets (used `filter()`).

The above are few of the implementations for the web app.

Kindly review the PR and provide feedback on the same.

Thanks.

### CODE REVIEW

File: src/redux/rockets/rocketSlice.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  rockets: [],
+  isLoading: true,
+};
+
+export const fetchRockets = createAsyncThunk(
+  'rockets/fetchRockets',
+  async (_, thunkAPI) => {
+    try {
+      const response = await axios.get('https://api.spacexdata.com/v4/rockets');
+      return response.data;
```
Comment (julie-ify) [inline src/redux/rockets/rocketSlice.js:16]: - [x] The rockets data from the API should be filtered. It is not all data that should be stored. You filtered the missions data but failed to do the same for the rockets. Kindly do this.

File: README.md [hunk 2]
```diff
@@ -63,174 +32,114 @@ For more information on the importance of a professional README for your reposit
...
-Example command:
-
-```sh
-  rails server
-```
---->
 
 ### Run tests
 
 To run tests, run the following command:
```
Comment (julie-ify) [inline README.md:110]: - [x] Kindly write unit tests for your components. You have no tests now.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @techmoves and @mailsg,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (techmoves) [reply README.md:110]: Thanks test added

File: src/redux/rockets/rocketSlice.js (hunk 1 above)
Comment (techmoves) [reply src/redux/rockets/rocketSlice.js:16]: Thanks it resolved!

**Review by alexander16108 — APPROVED** [review]
Hi @techmoves @mailsg , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Good code organization and working app ✔️
Descriptive README ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**